radiomics/shape.py

```diff
@@ -239,57 +235,38 @@ def getMaximum3DDiameterFeatureValue(self):
     r"""
     **8. Maximum 3D diameter**
 
-    Maximum 3D diameter is defined as the largest pairwise Euclidean distance between surface voxels in the ROI.
+    Maximum 3D diameter is defined as the largest pairwise Euclidean distance between tumor surface mesh
+    vertices.
 
     Also known as Feret Diameter.
-
-    .. warning::
-      This feature is only available when C Extensions are enabled
     """
-    if self.diameters is None:
-      self.diameters = self.calculateDiameters()
     return self.diameters[3]
 
   def getMaximum2DDiameterSliceFeatureValue(self):
     r"""
     **9. Maximum 2D diameter (Slice)**
 
-    Maximum 2D diameter (Slice) is defined as the largest pairwise Euclidean distance between tumor surface voxels in
-    the row-column (generally the axial) plane.
-
-    .. warning::
-      This feature is only available when C Extensions are enabled
+    Maximum 2D diameter (Slice) is defined as the largest pairwise Euclidean distance between tumor surface mesh
+    vertices in the row-column (generally the axial) plane.
     """
-    if self.diameters is None:
-      self.diameters = self.calculateDiameters()
     return self.diameters[0]
 
   def getMaximum2DDiameterColumnFeatureValue(self):
     r"""
     **10. Maximum 2D diameter (Column)**
 
-    Maximum 2D diameter (Column) is defined as the largest pairwise Euclidean distance between tumor surface voxels in
-    the row-slice (usually the coronal) plane.
-
-    .. warning::
-      This feature is only available when C Extensions are enabled
+    Maximum 2D diameter (Column) is defined as the largest pairwise Euclidean distance between tumor surface mesh
+    vertices in the row-slice (usually the coronal) plane.
     """
-    if self.diameters is None:
-      self.diameters = self.calculateDiameters()
     return self.diameters[1]
 
   def getMaximum2DDiameterRowFeatureValue(self):
     r"""
     **11. Maximum 2D diameter (Row)**
 
-    Maximum 2D diameter (Row) is defined as the largest pairwise Euclidean distance between tumor surface voxels in the
-    column-slice (usually the sagittal) plane.
-
-    .. warning::
-      This feature is only available when C Extensions are enabled
+    Maximum 2D diameter (Row) is defined as the largest pairwise Euclidean distance between tumor surface mesh
+    vertices in the column-slice (usually the sagittal) plane.
     """
-    if self.diameters is None:
-      self.diameters = self.calculateDiameters()
     return self.diameters[2]
 
   def getMajorAxisFeatureValue(self):
```

Comment:
Should this be `MajorAxisLength`? To me, "major axis" implies vector.

Reply:
@fedorov, good point! But I'll change that in a separate PR, as it is unrelated to the changes proposed here.